web/src/client/users.js

```diff
@@ -64,13 +64,12 @@ class UsersBaseClient {
    * @return {Promise<User>}
    */
   async getUser() {
-    const user = await this.client.get("/users/first");
-
-    if (user === null) {
+    const response = await this.client.get("/users/first");
+    if (!response.ok) {
+      console.log("Failed to get first user config: ", response);
       return { fullName: "", userName: "", password: "", autologin: false, data: {} };
```

Comment:
Later we should probably improve the error handling. This basically silently hides the error, users usually do no check the console logs.

The problem is when some of the defaults gets back to the server as a changed value. This can potentially loose some settings on the server accidentally if there is some temporary network problem (i.e. the GET call fails but POST later succeeds).

I think there should be some automatic retry with several attempts. If it still fails we should tell the user and probably do not allow to continue, or offer a full page reload to get into the consistent state...

Reply:
yeah, error reporting is something we can definitively improve.